src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorWorkspaceListener.cs

```diff
@@ -45,17 +50,38 @@ public void Dispose()
         _disposeTokenSource.Dispose();
     }
 
-    public void EnsureInitialized(Workspace workspace, string projectInfoFileName)
+    public void EnsureInitialized(Workspace workspace, string pipeName)
```

Comment:
Can this method be called from multiple threads?

Reply:
It shouldn't be. We could even throw if it's called more than once, but we chose to gracefully ignore. This is called from a factory on the Roslyn side.